server/base/src/main/java/org/apache/accumulo/server/metadata/ServerAmpleImpl.java

```diff
@@ -192,10 +192,9 @@ private BatchWriter createWriter(TableId tableId) {
     }
   }
 
-  public static Mutation createDeleteMutation(ServerContext context, TableId tableId,
-      String pathToRemove) {
-    Path path = context.getVolumeManager().getFullPath(tableId, pathToRemove);
-    Mutation delFlag = new Mutation(new Text(DeletesSection.encodeRow(path.toString())));
+  public static Mutation createDeleteMutation(String pathToRemove) {
+    String path = TabletFileUtil.validate(pathToRemove);
+    Mutation delFlag = new Mutation(new Text(DeletesSection.encodeRow(path)));
```

Comment:
The old code did a kind of normalization here against the volume manager's filesystem. I would hope that by the time it gets here, all callers would be providing a normalized string, but care should be taken if the parameter is provided from user input, that we normalize it first. I didn't track down all the callers to check this.

Reply:
Also, there's a lot of calls to `toString()` on the path to validate. This is the only method that calls validate without calling `toString()` first. However, many of this method's own callers call `toString()` first on the parameter. I think some of this could be simplified if everybody stopped calling `toString()`, and we kept the more concrete type, and only call `toString()` inside the validate method, if necessary. It simplifies a lot of the calling code.

Reply:
Which object do you think would be better?  The issue is that we have some use of Path, ByteSequence and String.  Calls to the ```createDeleteMutation()``` method are particularly hairy as there are a few spots where we parse the string and do string manipulation:
[gc](https://github.com/apache/accumulo/blob/3fd5cad92f9b63ac19e4466f3f2d5237b905262c/server/base/src/main/java/org/apache/accumulo/server/gc/GcVolumeUtil.java#L36)
[upgrader](https://github.com/apache/accumulo/blob/3276e788e8fa7ebf9c5f1fa4f37f49bad4eaeb3a/server/master/src/main/java/org/apache/accumulo/master/upgrade/Upgrader9to10.java#L434)

Reply:
I think you are pointing to an overall problem with a lot of the internal code.  One of my goals with the new ```TabletFile``` (similar to how we are trying to use Ample as much as possible) was to try and replace as much overuse of String and FileRef as we can.  We have a ```LogEntry``` which I think is good for WALs but we might need some more internal types for strings that aren't TabletFiles.

Reply:
That being said, I think a lot of that can be done in other PRs while the goal of this one was to remove some of the relative path code that was no longer needed in VolumeManager

Reply:
I'm happy to defer to future improvements, if what can be done here is limited.